server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java

```diff
@@ -81,28 +80,6 @@ public TableConfiguration(ServerContext context, TableId tableId, NamespaceConfi
         newDeriver(conf -> createCryptoService(conf, tableId, context.getCryptoFactory()));
   }
 
-  @Override
-  public String get(Property property) {
-    String value = _get(property);
-    if (value != null) {
-      return value;
-    }
-    AccumuloConfiguration parent = getParent();
-    if (parent != null) {
-      return parent.get(property);
-    }
-    return null;
-  }
-
-  @Nullable
-  private String _get(Property property) {
-    Map<String,String> propMap = getSnapshot();
-    if (propMap == null) {
```

Comment:
Compared this removed code w/ the code in ZooBasedConfiguration.  The code in ZooBasedConfiguration would throw a NPE if `getSnapshot()` returned null.  AFACIT this is the only difference.  This seems sketchy, was this a bug?  2.1 has the same behavior wondering if any change is needed there.

Reply:
It's not possible. It's always non-null. We even have a `@NonNull` annotation to signal that. The ZooBasedConfiguration didn't bother to check it for that reason, but this code was too far distant, so I guess it diverged a bit.